Kitodo-ImageManagementModule/src/main/java/org/kitodo/imagemanagementmodule/ImageManagementModule.java

```diff
@@ -0,0 +1,197 @@
+            logger.trace("{} successfully loaded", tempFile);
+            return buffer;
+        } finally {
+            logger.debug("Deleting {}", tempFile);
+            tempFile.delete();
+            logger.trace("Successfully deleted {}", tempFile);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.kitodo.api.imagemanagement.ImageManagementInterface#createDerivative(java.net.URI,
+     *      double, java.net.URI,
+     *      org.kitodo.api.imagemanagement.ImageFileFormat)
+     */
+    @Override
+    public boolean createDerivative(URI sourceUri, double factor, URI resultUri, ImageFileFormat format)
+            throws IOException {
+
+        if (!new File(sourceUri).exists()) {
+            throw new FileNotFoundException("sourceUri must exist: " + sourceUri.getRawPath());
+        }
+        if (Double.isNaN(factor)) {
+            throw new IllegalArgumentException("factor must be a number, but was " + Double.toString(factor));
+        }
+        if (factor <= 0.0) {
+            throw new IllegalArgumentException("factor must be > 0.0, but was " + Double.toString(factor));
+        }
+        if (resultUri == null) {
+            throw new NullPointerException("resultUri must not be null");
+        }
+
+        ImageConverter imageConverter = new ImageConverter(sourceUri);
+        imageConverter.addResult(resultUri, format).resize(factor);
+        logger.info("Creating derivative from {} as {}, format {}, factor {}%", sourceUri, resultUri, format,
+            100 * factor);
+        imageConverter.run();
+        return new File(resultUri).exists();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.kitodo.api.imagemanagement.ImageManagementInterface#changeDpi(java.net.URI,
+     *      int)
+     */
+    @Override
+    public Image changeDpi(URI sourceUri, int dpi) throws IOException {
+        if (!new File(sourceUri).exists()) {
+            throw new FileNotFoundException("sourceUri must exist: " + sourceUri.getRawPath());
+        }
+        if (dpi <= 0) {
+            throw new IllegalArgumentException("dpi must be > 0, but was " + Integer.toString(dpi));
+        }
+
+        File tempFile = File.createTempFile("dpiChangedImage-", RAW_IMAGE_FORMAT, TMPDIR);
+        try {
+            tempFile.deleteOnExit();
+            URI imageUri = tempFile.toURI();
+            ImageConverter imageConverter = new ImageConverter(sourceUri);
+            imageConverter.addResult(imageUri).resizeToDpi(dpi);
+
+            logger.info("Resizing {} as {} to {} DPI", sourceUri, tempFile, dpi);
+            imageConverter.run();
+
+            logger.trace("Loading {}", tempFile);
+            Image buffer = ImageIO.read(tempFile);
+            logger.trace("{} successfully loaded", tempFile);
+            return buffer;
+        } finally {
+            logger.debug("Deleting {}", tempFile);
+            tempFile.delete();
+            logger.trace("Successfully deleted {}", tempFile);
```

Comment:
Same as in other method: you are logging a successful deleting without using the return value of `delete` method.